### PR TITLE
dev-util/hyprwayland-scanner: new package, add 0.2.0, 9999

### PR DESCRIPTION
**Pull request by jthro**
Signed-off-by: jthro <jethro.rosette@outlook.com>

dev-util/hyprwayland-scanner: new package, add 0.2.0, 9999, bug #930393

hyprwayland-scanner is required to build newer versions of the Hyprland window manager (see: https://github.com/hyprwm/Hyprland/pull/5657)

Bug: https://bugs.gentoo.org/930393

### CODE REVIEW

**Comment by jthro** [comment]
I don't think I got the metadata.xml quite correct. Any pointers as to how to fix would be appreciated.

File: dev-util/hyprwayland-scanner/hyprwayland-scanner-0.2.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+DESCRIPTION="A Hyprland implementation of wayland-scanner, in and for C++"
+
+HOMEPAGE="https://github.com/hyprwm/hyprwayland-scanner/"
+
+inherit cmake
+if [[ ${PV} == 9999 ]] ; then
+	EGIT_REPO_URI="https://github.com/hyprwm/hyprwayland-scanner.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/hyprwm/hyprwayland-scanner/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
```
Comment (MrRoy) [inline dev-util/hyprwayland-scanner/hyprwayland-scanner-0.2.0.ebuild:15]: Simplify by removing `refs/tags/` and add `.gh` to distfile name
```suggestion
	SRC_URI="https://github.com/hyprwm/hyprwayland-scanner/archive/v${PV}.tar.gz -> ${P}.gh.tar.gz"
```

File: dev-util/hyprwayland-scanner/hyprwayland-scanner-0.2.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DESCRIPTION="A Hyprland implementation of wayland-scanner, in and for C++"
+
+HOMEPAGE="https://github.com/hyprwm/hyprwayland-scanner/"
+
+inherit cmake
```
Comment (MrRoy) [inline dev-util/hyprwayland-scanner/hyprwayland-scanner-0.2.0.ebuild:10]: inherit goes before DESCRIPTION, also remove newline between DESCRIPTION & HOMEPAGE

```suggestion
EAPI=8

inherit cmake

DESCRIPTION="A Hyprland implementation of wayland-scanner, in and for C++"
HOMEPAGE="https://github.com/hyprwm/hyprwayland-scanner/"
```

File: dev-util/hyprwayland-scanner/hyprwayland-scanner-0.2.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+if [[ ${PV} == 9999 ]] ; then
+	EGIT_REPO_URI="https://github.com/hyprwm/hyprwayland-scanner.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/hyprwm/hyprwayland-scanner/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64"
+
+fi
```
Comment (MrRoy) [inline dev-util/hyprwayland-scanner/hyprwayland-scanner-0.2.0.ebuild:18]: Remove newline
```suggestion
	KEYWORDS="~amd64"
fi
```

File: dev-util/hyprwayland-scanner/hyprwayland-scanner-0.2.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+LICENSE="BSD"
+
+SLOT="0"
```
Comment (MrRoy) [inline dev-util/hyprwayland-scanner/hyprwayland-scanner-0.2.0.ebuild:22]: remove newline here too

```suggestion
LICENSE="BSD"
SLOT="0"
```

**Review by MrRoy — CHANGES_REQUESTED** [review]
@jthro
Left my comments on 0.2.0 ebuild but they apply also to 0.3.1 and 9999

Few more points:
- Can you add me as a maintainer to metadata.xml? Julien Roy <julien@jroy.ca> 
- Is there any reason to keep 0.2.0 now that 0.3.1 is released? I would just drop 0.2.0 since it's a new package
- I'm not sure including the live version is useful, it sounds more like a maintenance chore and I don't see a use case for it. Let me know if you disagree. (we can keep the `if [[ ${PV} == 9999 ]]` part though.)
- Can you include a commit to update the -9999 Hyprland ebuild to add this new dependency?
- In your commit message, instead of `Bug:`, use `Closes:` tag

Another note:
Upstream says that this needs a compiler that supports C++23. You could copy [the pkg_setup() phase from the gui-wm/hyprland ebuild](https://github.com/gentoo/gentoo/blob/12b9284c3a9b57403d8b9e6f5b7be4da383c4338/gui-wm/hyprland/hyprland-0.39.1-r1.ebuild#L104)

**Comment by jthro** [comment]
I have resolved all of these comments, however it doesn't seem to successfully compile at the moment and complains about an invalid type conversion. I get the impression this might be an upstream problem.

**Comment by dlan17** [comment]
@jthro you can squash the commits and do a git force push, please keep at least one commit per package or more commits at  detaiedl logic..

File: dev-util/hyprwayland-scanner/hyprwayland-scanner-0.3.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Copyright 1999-2024 Gentoo Authors
```
Comment (dlan17) [inline dev-util/hyprwayland-scanner/hyprwayland-scanner-0.3.1.ebuild:1]: for new package, just start at current year, so it should be1 "Copyright 2024 Gentoo Authors"

File: dev-util/hyprwayland-scanner/hyprwayland-scanner-0.3.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+	EGIT_REPO_URI="https://github.com/hyprwm/hyprwayland-scanner.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/hyprwm/hyprwayland-scanner/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64"
+
+fi
+
+LICENSE="BSD"
+
```
Comment (dlan17) [inline dev-util/hyprwayland-scanner/hyprwayland-scanner-0.3.1.ebuild]: I think there are some blank lines.. please use "pkgcheck scan --commits $(git merge-base HEAD origin/master)..HEAD" to check..

for commit msg, always keep your signed-off at bottom line.

File: dev-util/hyprwayland-scanner/hyprwayland-scanner-0.3.1.ebuild [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+DESCRIPTION="A Hyprland implementation of wayland-scanner, in and for C++"
+
+HOMEPAGE="https://github.com/hyprwm/hyprwayland-scanner/"
+
+inherit cmake
+if [[ ${PV} == 9999 ]] ; then
+	EGIT_REPO_URI="https://github.com/hyprwm/hyprwayland-scanner.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/hyprwm/hyprwayland-scanner/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
```
Comment (dlan17) [inline dev-util/hyprwayland-scanner/hyprwayland-scanner-0.3.1.ebuild]: I'd suggest to use ${P}.gh.tar.gz, as it's a tag release from github.. just as a convention..

**Comment by jthro** [comment]
I have made an absolute dog's breakfast of the commit history. Apologies. Fixing now.

**Comment by jthro** [comment]
I have resolved these issues. Apologies for the commit history I made some errors but they should all be solved now and the commits should be in the correct format.

**Comment by dlan17** [comment]
thanks, I've pushed..  squash first two commits, and fix the missing toolchain-funcs warning

**Pull request closed without merging**